src/StructChildField.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {connect} from 'react-redux';
+import {
+  getFieldName, getLabel,
+  getNestedBlockDefinition,
+  isRequired
+} from './processing/utils';
+import FieldInput from './FieldInput';
+
+
+@connect((state, props) => {
+  const {fieldId, parentBlockId, type} = props;
+  const blocks = state[fieldId].blocks;
+  const parentBlock = blocks[parentBlockId];
+  const blockId = parentBlock.value.find(
+    childBlockId => blocks[childBlockId].type === type);
+  return {
+    blockDefinition: getNestedBlockDefinition(state, fieldId, blockId),
+    blockId,
+  };
+})
+class StructChildField extends React.Component {
+  static propTypes = {
+    fieldId: PropTypes.string.isRequired,
+    parentBlockId: PropTypes.string.isRequired,
+    type: PropTypes.string.isRequired,
+  };
+
+  render() {
+    const {fieldId, blockId, blockDefinition} = this.props;
+    return (
+      <div className={classNames('field',
```

Comment:
This is done to match Wagtail’s classes. BEMifying this would require a significant Wagtail rewrite.
So it indeed has to be changed in the future, but for now I keep it as is.